tests/tests_app/utilities/test_proxies.py

```diff
@@ -790,3 +792,27 @@ def test_bi_directional_proxy_filtering():
     app = LightningApp(FlowDrive())
     app.root.run()
     assert app._extract_vars_from_component_name(app.root.w.name, app.state) == {}
+
+
+def test_signal_handler_compose(monkeypatch):
+
+    examples = []
+
+    handler = SignalHandlerCompose(signal.signal)
+
+    monkeypatch.setattr(signal, "signal", handler)
+
+    def handler_fn(signum, handler):
+        nonlocal examples
+        examples.append("handler_fn")
+
+    def handler_fn2(signum, handler):
+        nonlocal examples
+        examples.append("handler_fn2")
+
+    signal.signal(signal.SIGTERM, handler_fn)
+    signal.signal(signal.SIGTERM, handler_fn2)
+
+    handler.run(signal.SIGTERM, None)
+
+    assert examples == ["handler_fn", "handler_fn2"]
```

Comment:
We can see what's wrong here: The signal handlers are a global setting. The expectation is that if you call

signal.signal(x, handler) 

then it should replace the handler that was already there. In this PR, Lightning breaks this contract in all other libraries, since it patches signal.signal globally. 

I believe the intended use should be:

```
signal.signal(signum, HandlerCompose(handler1, handler2, ...))
```

Reply:
So the problem we are trying to solve here is that uvloop overwrites the signal handler we set. So if we don't patch `signal.signal`, as soon as we run `uvcorn.run`, it overwrites the handler and hence `ctrl-c` would never terminate the work process correctly

Reply:
Then I suggest we only patch it for uvicorn/ in uvicorn